corehq/apps/hqadmin/views.py

```diff
@@ -395,6 +398,10 @@ class AdminRestoreView(TemplateView):
     template_name = 'hqadmin/admin_restore.html'
 
     @method_decorator(require_superuser)
+    def dispatch(self, request, *args, **kwargs):
+        super(AdminRestoreView, self).dispatch(*args, **kwargs)
```

Comment:
Would this cause an error since you aren't passing request to dispatch? Also should you  have  `return` ?